### PR TITLE
Throwing when goldctl cannot authorize/initialize

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This PR adds throw conditions for the `auth` and `init` use cases of `goldctl`.  Since we use `passfail` mode for Gold, we have silenced errors that may be due to intended golden file changes. `auth` and `init` will not throw for intended changes, so should be safe to use to detect unwanted errors from Gold.

This means that master will break if an error is picked up from `goldctl`. We automatically pick up the latest version of `goldctl` from cipd for each build. There has been discussion of including it in the Dockerfile in #44786 to mitigate this exposure and reduce the work we are doing for each build.

## Related Issues

Part of detecting issues like #45699

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Piinks** [comment]
FYI @kjlubick

**Comment by kjlubick** [comment]
\o/

**Review by goderbauer — COMMENTED** [review]
Can you add tests?

File: packages/flutter_goldens/test/flutter_goldens_test.dart [hunk 1]
```diff
@@ -66,20 +67,91 @@ void main() {
...
-          .thenAnswer((_) => Future<ProcessResult>
-            .value(ProcessResult(123, 0, '', '')));
-        await skiaClient.auth();
-
-        verifyNever(process.run(
-            captureAny,
-            workingDirectory: captureAnyNamed('workingDirectory'),
-        ));
-      });
+    test(' auth performs minimal work if already authorized', () async {
```
Comment (goderbauer) [inline packages/flutter_goldens/test/flutter_goldens_test.dart]: nit: do we need the leading space here?

Comment (goderbauer) [reply packages/flutter_goldens/test/flutter_goldens_test.dart]: Same below.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by Piinks** [comment]
The status of `customer_testing-macos` and `framework_tests-misc-macos` are green on Cirrus, despite the red status shown here.

File: packages/flutter_goldens/test/flutter_goldens_test.dart (hunk 1 above)
Comment (Piinks) [inline packages/flutter_goldens/test/flutter_goldens_test.dart]:
```suggestion
    test('auth performs minimal work if already authorized', () async {
```

File: packages/flutter_goldens/test/flutter_goldens_test.dart [hunk 2]
```diff
@@ -66,20 +67,91 @@ void main() {
...
+        .value(ProcessResult(123, 0, '', '')));
+      await skiaClient.auth();
+
+      verifyNever(process.run(
+        captureAny,
+        workingDirectory: captureAnyNamed('workingDirectory'),
+      ));
+    });
+
+    test(' throws for error state from auth', () async {
```
Comment (Piinks) [inline packages/flutter_goldens/test/flutter_goldens_test.dart]:
```suggestion
    test('throws for error state from auth', () async {
```

**Pull request merged**